src/plugins/telemetry_collection_manager/server/plugin.ts

```diff
@@ -134,6 +128,39 @@ export class TelemetryCollectionManagerPlugin
     }
   }
 
+  /**
+   * Returns the ES client scoped to the requester or Kibana's internal user
+   * depending on whether the request is encrypted or not:
+   * If the request is unencrypted, we intentionally scope the results to "what the user can see".
+   * @param config {@link StatsGetterConfig}
+   * @private
+   */
+  private getElasticsearchClient(config: StatsGetterConfig): ElasticsearchClient | undefined {
+    return config.unencrypted
+      ? this.elasticsearchClient?.asScoped(config.request).asCurrentUser
+      : this.elasticsearchClient?.asInternalUser;
+  }
+
+  /**
+   * Returns the SavedObjects client scoped to the requester or Kibana's internal user
+   * depending on whether the request is encrypted or not:
+   * If the request is unencrypted, we intentionally scope the results to "what the user can see"
+   * @param config {@link StatsGetterConfig}
+   * @private
+   */
+  private getSavedObjectsClient(config: StatsGetterConfig): SavedObjectsClientContract | undefined {
+    if (config.unencrypted) {
+      // Intentionally using the scoped client here to make use of all the security wrappers.
+      // It also returns spaces-scoped telemetry.
+      return this.savedObjectsService?.getScopedClient(config.request);
+    } else if (this.savedObjectsService) {
+      // Wrapping the internalRepository with the `TelemetrySavedObjectsClient`
+      // to ensure some best practices when collecting "all the telemetry"
+      // (i.e.: `.find` requests should query all spaces)
+      return new TelemetrySavedObjectsClient(this.savedObjectsService.createInternalRepository());
```

Comment:
How is this implementation different to what it was before?
Other than refactoring (which makes it cleaner, thanks!), I cannot see any obvious differences.